src/main/java/io/vertx/amqp/bridge/impl/AmqpProducerImpl.java

```diff
@@ -0,0 +1,141 @@
+/*
+* Copyright 2016 the original author or authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package io.vertx.amqp.bridge.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.VertxException;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.eventbus.MessageProducer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonSender;
+
+public class AmqpProducerImpl implements MessageProducer<JsonObject> {
```

Comment:
Same comment than AmqpConsumerImpl about synchronization. For example the drainHandler could be set from a non Vert.x thread (a main) but the callback for draining drain would likely be done by a Vert.x thread (proton).